src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -234,6 +235,15 @@ protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
         }
     }
 
+    @Override
+    protected @Nonnull String getSuccessfulDestination() {
+        FolderComputation<I> computation = getComputation();
+        if (computation != null) {
```

Comment:
So does this actually work? My guess is that it does not, since `scheduleBuild()` has not even been called by the time this is called inside `super.doConfigSubmit`.

And even if you called `scheduleBuild()` first, that would not suffice, since that merely adds an item to the queue, which may not be scheduled immediately. Typically it will be scheduled very soon, since it is just a flyweight master task; potentially there could be a `QueueTaskDispatcher` or a `ComputedFolder.getCauseOfBlockage` override which prevents that, though I am not aware of any currently. At any rate, you must account for the possibility that the `FolderComputation` has not started instantly.

I suspect that what you need to do is delete the `doConfigSubmit` override, then have this method do something like

```java
Queue.Item item = scheduleBuild2(0);
if (item != null) {
    try {
        FolderComputation<?> computation = item.getFuture().getStartCondition().get(1, TimeUnit.SECONDS);
        return computation.getSearchUrl() + "console";
    } catch (TimeoutException x) {
        // fine, ignore
    } catch (InterruptedException|ExecutionException x) {
        // log?
    }
}
return super.getSuccessfulDestination(); // fallback
```

(Even better would be to check to see if the computation _finishes_ very quickly, and if it does so and produces at least one child, redirect to `.` since we can guess that all went more or less OK and the user is more interested in the children than the log. This would not typically be the case for multibranch projects, which generally make network calls that can take a while, but would often be true for things like [JENKINS-33275](https://issues.jenkins-ci.org/browse/JENKINS-33275).)